pytket/qir/conversion/qirgenerator.py

```diff
@@ -731,6 +791,153 @@ def conv_classicalexpbox(self, op: ClassicalExpBox, args: list) -> None:
+                    reg_posn,
+                    cmd_args,
+                    expr_op in _TK_CLEXPR_OP_WITH_REG_ARGS,
+                )[1]
+                for expr_arg in arg.args
+            ]
+            match expr_op:
+                case ClOp.BitAnd | ClOp.BitOr | ClOp.BitXor | ClOp.BitEq | ClOp.BitNeq:
+                    return True, _TK_CLEXPR_OP_TO_PYQIR[expr_op](self.module.builder)(
+                        *ssa_args
+                    )
+                case ClOp.BitNot:
+                    # Implemented as x --> 1 - x
+                    assert len(ssa_args) == 1
+                    return True, self.module.builder.sub(
+                        pyqir.const(self.qir_bool_type, 1), ssa_args[0]
+                    )
+                case ClOp.BitZero:
+                    assert len(ssa_args) == 0
+                    return True, pyqir.const(self.qir_bool_type, 0)
+                case ClOp.BitOne:
+                    assert len(ssa_args) == 0
+                    return True, pyqir.const(self.qir_bool_type, 1)
+                case (
+                    ClOp.RegAnd
+                    | ClOp.RegOr
+                    | ClOp.RegXor
+                    | ClOp.RegEq
+                    | ClOp.RegNeq
+                    | ClOp.RegLt
+                    | ClOp.RegGt
+                    | ClOp.RegLeq
+                    | ClOp.RegGeq
+                    | ClOp.RegAdd
+                    | ClOp.RegSub
+                    | ClOp.RegMul
+                    | ClOp.RegLsh
+                    | ClOp.RegRsh
+                ):
+                    assert len(ssa_args) == 2
+                    return False, _TK_CLEXPR_OP_TO_PYQIR[expr_op](self.module.builder)(
+                        *ssa_args
+                    )
+                case ClOp.RegNot:
+                    # Implemented as x --> 2^self.int_size - 1 - x
+                    assert len(ssa_args) == 1
+                    return False, self.module.builder.sub(
+                        pyqir.const(self.qir_int_type, (1 << self.int_size) - 1),
+                        ssa_args[0],
+                    )
+                case ClOp.RegZero:
+                    assert len(ssa_args) == 0
+                    return False, pyqir.const(self.qir_int_type, 0)
+                case ClOp.RegOne:
+                    # Sets all bits in the register to 1
+                    assert len(ssa_args) == 0
+                    return False, pyqir.const(
+                        self.qir_int_type, (1 << self.int_size) - 1
+                    )
+                case ClOp.RegNeg:
+                    # Implemented as x --> 0 - x
+                    assert len(ssa_args) == 1
+                    return False, self.module.builder.sub(
+                        pyqir.const(self.qir_int_type, 0), ssa_args[0]
+                    )
+                case ClOp.RegDiv | ClOp.RegPow:
+                    raise ValueError(f"Classical operation {expr_op} not supported")
```

Comment:
Do we want to link the issues here?

Reply:
I'll insert a comment with the issue URL.